contrib/checkpoint/checkpoint-restore-cri-test.sh

```diff
@@ -0,0 +1,213 @@
+	ctr_id=$(crictl create "$pod_id" "$TESTDATA"/container_sleep.json "$TESTDATA"/sandbox_config.json)
+	echo "$ctr_id"
+	echo -n "--> Start container: "
+	crictl start "$ctr_id"
+	echo -n "--> Checkpointing container: "
+	crictl -t 10s checkpoint --export="$TESTDIR"/cp.tar "$ctr_id"
+	echo "--> Cleanup container: "
+	crictl rm -f "$ctr_id" | sed 's/^/----> \t/'
+	echo "--> Cleanup pod: "
+	crictl rmp -f "$pod_id" | sed 's/^/----> \t/'
+	echo "--> Cleanup pod: "
+	crictl rmi "${TEST_IMAGE}" | sed 's/^/----> \t/'
+	# Change cgroup of new sandbox
+	RESTORE_POD_JSON=$(mktemp)
+	jq ".linux.cgroup_parent=\"different_cgroup_789\"" "$TESTDATA"/sandbox_config.json >"$RESTORE_POD_JSON"
+	echo -n "--> Start pod: "
+	pod_id=$(crictl runp "$RESTORE_POD_JSON")
+	echo "$pod_id"
+	# Replace original container with checkpoint image
+	RESTORE_JSON=$(mktemp)
+	# Convert tar checkpoint archive to OCI image
+	echo "--> Convert checkpoint archive $TESTDIR/cp.tar to OCI image 'checkpoint-image:latest': "
+	newcontainer=$(buildah from scratch)
+	echo -n "----> Add checkpoint archive to new OCI image: "
+	buildah add "$newcontainer" "$TESTDIR"/cp.tar /
+	echo "----> Add checkpoint annotation to new OCI image: "
+	buildah config --annotation=org.criu.checkpoint.container.name=test "$newcontainer"
+	echo "----> Save new OCI image: "
+	buildah commit -q "$newcontainer" checkpoint-image:latest 2>&1 | sed 's/^/------> \t/'
+	echo "----> Cleanup temporary images: "
+	buildah rm "$newcontainer" | sed 's/^/------> \t/'
+	# Export OCI image to disk
+	echo "----> Export OCI image to disk: "
+	podman image save -q --format oci-archive -o "$TESTDIR"/oci.tar localhost/checkpoint-image:latest | sed 's/^/------> \t/'
+	echo "----> Cleanup temporary images: "
+	buildah rmi localhost/checkpoint-image:latest | sed 's/^/------> \t/'
+	# Remove potentially old version of the checkpoint image
+	echo "----> Cleanup potential old copies: "
+	../../bin/ctr -n k8s.io images rm --sync localhost/checkpoint-image:latest 2>&1 | sed 's/^/------> \t/'
+	# Import image
+	echo "----> Import new image: "
+	../../bin/ctr -n k8s.io images import "$TESTDIR"/oci.tar 2>&1 | sed 's/^/------> \t/'
+	jq ".image.image=\"localhost/checkpoint-image:latest\"" "$TESTDATA"/container_sleep.json >"$RESTORE_JSON"
+	echo -n "--> Create container from checkpoint: "
+	ctr_id=$(crictl -t 10s create "$pod_id" "$RESTORE_JSON" "$RESTORE_POD_JSON")
+	echo "$ctr_id"
+	rm -f "$RESTORE_JSON" "$RESTORE_POD_JSON"
+	echo -n "--> Start container from checkpoint: "
+	crictl start "$ctr_id"
+	# Cleanup
+	echo "--> Cleanup images: "
+	../../bin/ctr -n k8s.io images rm localhost/checkpoint-image:latest | sed 's/^/----> \t/'
+	echo "--> Cleanup images: "
+	crictl rmi "${TEST_IMAGE}" | sed 's/^/----> \t/'
+	echo "--> Deleting all pods: "
+	crictl -t 5s rmp -fa | sed 's/^/----> \t/'
+	SUCCESS=1
+}
+
+cat >"${TESTDIR}/config.toml" <<EOF
+version = 3
+[plugins.'io.containerd.cri.v1.runtime'.containerd.runtimes.runc]
```

Comment:
nit: "runc" may cause a confusion when `TEST_RUNTIME` is set to non-runc

Reply:
Not sure what to do here.

Reply:
Maybe just call it "test-runtime"?

Reply:
Ah, good. Changed.